pkgs/by-name/ne/nerd-fonts/update.js

```diff
@@ -0,0 +1,72 @@
+const RELEASE_VERSION_FILENAME = "release-version.json";
+const RELEASE_INFO_URL = "https://api.github.com/repos/ryanoasis/nerd-fonts/releases/latest";
+const PKGS_INFO_FILENAME = "fonts.json";
+
+const getFontsInfoUrl = (tag) => `https://raw.githubusercontent.com/ryanoasis/nerd-fonts/refs/tags/${tag}/bin/scripts/lib/fonts.json`;
+
+const fetchJSON = (url) => JSON.parse(std.urlGet(url).split("\r\n").at(-1));
+
+const convertAttrName = (formerAttrName) => {
+    if (/^\d/u.test(formerAttrName)) {
+        return "_" + formerAttrName;
+    } else {
+        return formerAttrName;
+    }
+};
+
+const convertVersion = (formerVersion, dateString) => {
+    if (/^\d/u.test(formerVersion)) {
+        return formerVersion;
+    } else {
+        return "0-unstable-" + dateString;
+    }
+};
+
+let formerReleaseVersion = JSON.parse(std.loadFile(RELEASE_VERSION_FILENAME));
+let releaseInfo = fetchJSON(RELEASE_INFO_URL);
+let tag = releaseInfo.tag_name;
+let releaseVersion = tag.split("v")[1];
+
+if (releaseVersion === formerReleaseVersion) {
+    std.exit(1);
+} else {
+    let fonts = fetchJSON(getFontsInfoUrl(tag)).fonts;
+
+    let pkgsInfo = fonts.map(font => {
+        let pname = font.caskName.toLowerCase();
+
+        let url = releaseInfo.assets
+            .find(asset => asset.name === font.folderName + ".tar.xz")
+            .browser_download_url;
+
+        let sha256 = (() => {
+            let f = std.popen(`nix-prefetch-url ${url} --unpack`, "r");
+            let result = f.getline();
+            f.close();
+            return result;
+        })();
+
+        return ({
+            pname: pname,
+            attrName: convertAttrName(pname),
+            dirName: font.patchedName.replaceAll(/\s+/gu, ""),
+            version: convertVersion(font.version, releaseInfo.published_at.split("T")[0]),
+            longDescription: font.description,
+            license: font.licenseId,
+            url: url,
+            sha256: sha256,
```

Comment:
Hmm maybe use the "Reserved Font Name" as an attribute name? That is the most important attribute of the fonts - as it is the one you'd be using in configuration files.

Reply:
Urh... According to [SIL Open Font License](https://openfontlicense.org/):

> [If the font has RFNs you may need to change the font name (both internal name fields and anything a user might see) to avoid using any RFNs.
](https://openfontlicense.org/how-to-modify-ofl-fonts/)

> [If any RFNs are declared for a font, then Modified Version(s) of that font made by anyone other than the copyright holder cannot use those RFNs in the font name.](https://openfontlicense.org/ofl-reserved-font-names/)

So I think "for configuration files" might not be the intended usage of RFNs. And it might even be expected not to use these names at all except for copyright declarations.

Reply:
> Urh... According to [SIL Open Font License](https://openfontlicense.org/):
> 
> > [If the font has RFNs you may need to change the font name (both internal name fields and anything a user might see) to avoid using any RFNs.
> > ](https://openfontlicense.org/how-to-modify-ofl-fonts/)
> 
> > [If any RFNs are declared for a font, then Modified Version(s) of that font made by anyone other than the copyright holder cannot use those RFNs in the font name.](https://openfontlicense.org/ofl-reserved-font-names/)
> 
> So I think "for configuration files" might not be the intended usage of RFNs. And it might even be expected not to use these names at all except for copyright declarations.

What? How is that related to licensing? I don't get it. Almost every program that can customize its fonts is using this RFN in their configuration files. Examples include:

- Alacritty (terminal emulator): https://alacritty.org/config-alacritty.html#font
- [Qalculate-gtk](https://github.com/Qalculate/qalculate-gtk): Can't prove that from the source code, but can quote my personal configuration file:
  `custom_application_font=Source Sans 3 11`
- `mlterm`: https://github.com/arakiken/mlterm/blob/rel-3_9_1/etc/aafont

Or did I understand the term RFN incorrectly?

Reply:
I'm sorry; just didn't notice these ... practical concerns. I'll try to supplement them.